projects/client/RabbitMQ.Client/src/client/api/IConnection.cs

```diff
@@ -178,6 +178,15 @@ public interface IConnection : NetworkConnection, IDisposable
 
         event EventHandler<EventArgs> ConnectionUnblocked;
 
+        /// <summary>
+        /// This method updates the secret used to authenticate this connection.
+        /// It is used when secrets have an expiration date and need to be renewed,
+        /// like OAuth 2 tokens.
+        /// </summary>
+        /// <param name="newSecret">The new secret.</param>
+        /// <param name="reason">The reason for the secret update.</param>
+        void UpdateSecret(string newSecret, string reason);
```

Comment:
`update-secret` is synchronous and returns `update-secret-ok`. I know the result does not contain critical information, but I'm just pointing this out, in case the method should reflect this.

Reply:
In the existing code, it seems like:

- No method in IConnection returns a *Ok result. 
- One method in IModel returns an *Ok result, namely QueueDeclare. QueueDeclareOk contains the QueueName, which makes sense for auto-named queues. 

In the case of SecretOk, as you said, there is no information. When there is an error, there is an exception. For me, it works without a SecretOk result. But it's ultimately your call.

Reply:
In the existing code, it seems like:

- No method in IConnection returns a *Ok result.
- One method in IModel returns an *Ok result, namely QueueDeclare. QueueDeclareOk contains the QueueName, which makes sense for auto-named queues.

In the case of SecretOk, as you said, there is no information. When there is an error, there is an exception. For me, it works without a SecretOk result. But it's ultimately your call.

Reply:
Uhm, yeah, API methods for protocol methods such as `exchange.bind` return a `void` as well. It's fine if `UpdateSecret` does the same for now.

Reply:
I think we should at least keep the synchronous semantics. So I just want to make sure that even though the method returns `void`, this is not "fire-and-forget" and a call does not return immediately.

Reply:
That part I agree with.

Reply:
Yes, that is the current behavior of UpdateSecret.

Reply:
I can also confirm RabbitMQ 3.8 does not complain when calling UpdateSecret with the default auth backend. I added a test for this.